src/capi.rs

```diff
@@ -441,23 +441,30 @@ pub unsafe extern fn rav1e_config_parse_int(
   }
 }
 
+fn context_new_internal(cfg: &rav1e::Config) -> Result<Context, rav1e::EncoderStatus> {
+  let enc = &cfg.enc;
+  Ok(Context {
+    ctx: match enc.bit_depth {
+      8 => EncContext::U8(cfg.new_context()?),
+      _ => EncContext::U16(cfg.new_context()?)
+    },
+    last_err: None
+  })
+}
+
 /// Generate a new encoding context from a populated encoder configuration
 ///
 /// Multiple contexts can be generated through it.
+/// Returns Null if context creation failed, e.g. by passing
+/// an invalid Config.
 #[no_mangle]
 pub unsafe extern fn rav1e_context_new(cfg: *const Config) -> *mut Context {
   let cfg = &(*cfg).cfg;
-  let enc = &cfg.enc;
-
-  let ctx = Context {
-    ctx: match enc.bit_depth {
-      8 => EncContext::U8(cfg.new_context()),
-      _ => EncContext::U16(cfg.new_context())
-    },
-    last_err: None
-  };
-
-  Box::into_raw(Box::new(ctx))
+  if let Ok(ctx) = context_new_internal(cfg) {
+    Box::into_raw(Box::new(ctx))
+  } else {
+    std::ptr::null_mut()
```

Comment:
Does this properly work with last_err?

Reply:
I don't think so. I can make it work.

Reply:
@lu-zero I don't actually know how to fix this. Suggestion?

Reply:
Either we add a last_err to Config and more api surface or it is fine to just return NULL.

I would just return NULL.